### ISSUE TITLE
Schema Reference not Found

### ISSUE THREAD

**Issue by sujithre** [Azure/azure-api-management-devops-resource-kit#738]
Hi All,

i am Using Azure Devops Resource Kit to deploy my APIS into the new APIM Instance. when i am deploying the API, for some of the Operations  i get the below Error

2022-06-02T07:14:28.4933958Z ##[error]Details:

2022-06-02T07:14:28.4935158Z ##[error]ValidationError: Operation references schema that does not exist.

2022-06-02T07:14:28.4936264Z ##[error]ValidationError: Operation references schema that does not exist.

I checked the Resource Group Deployment Logs and few of the operations are failing with the below Error

 
```json
{

"status": "Failed",

"error": {

"code": "ValidationError",

"message": "One or more fields contain incorrect values:",

"details": [

{

"code": "ValidationError",

"target": "operation 'GET'-'/resource?param1={param1}' parameter 'param2' schema XXXX",

"message": "Operation references schema that does not exist."

}

]

}

}
```
 

Can you Please help me is there anything i am doing it wrong here?

**Comment by DeagleGross** [comment]
Hello @sujithre,
can you, please, take advantage of  issue template and submit important information?
- resource-kit version you are using
- your input configuration to `Extractor` or `Creator` - whatever you are using
- include generated template part with operation, that is failing to deploy. So that we can see what is wrong with that `ApiOperation` resource.
- include `parameters` section, to see if that `param2` is missing

**Comment by sujithre** [comment]
Hi Deagle
Please fidn the below
resource-kit version you are using? v0.7
your input configuration to Extractor or Creator - Extractor
include generated template part with operation, that is failing to deploy. So that we can see what is wrong with that ApiOperation resource.
include parameters section, to see if that param2 is missing
 {
      "apiVersion": "2021-08-01",
      "type": "Microsoft.ApiManagement/service/apis/operations",
      "name": "[concat(parameters('apimServiceName'), '/echo-api/retrieve-resource')]",
      "dependsOn": [
        "[resourceId('Microsoft.ApiManagement/service/apis', parameters('apimServiceName'), 'echo-api')]"
      ],
      "properties": {
        "displayName": "Retrieve resource",
        "method": "GET",
        "urlTemplate": "/resource?param1={param1}",
        "description": "A demonstration of a GET call on a sample resource. It is handled by an \"echo\" backend which returns a response equal to the request (the supplied headers and body are being returned as received).",
        "templateParameters": [
          {
            "name": "param1",
            "description": "A sample parameter that is required and has a default value of \"sample\".",
            "type": "string",
            "defaultValue": "sample",
            "required": true,
            "values": [
              "sample"
            ]
          }
        ],
        "request": {
          "headers": [],
          "queryParameters": [
            {
              "description": "Another sample parameter, set to not required.",
              "name": "param2",
              "required": false,
              "schemaId": "6273a22d217d200f783808ae",
              "type": "number",
              "typeName": "ResourceGetRequest-1"
            }
          ],
          "representations": []
        },
        "responses": [
          {
            "description": "Returned in all cases.",
            "headers": [],
            "representations": [],
            "statusCode": 200
          }
        ]
      }
    }
    Do you want me to attach apis.Template.json which got extracted from Extractor?

**Comment by DeagleGross** [comment]
@sujithre can you please try the latest beta release? https://github.com/Azure/azure-api-management-devops-resource-kit/releases/tag/1.0.0-beta.5

V07 is an old version, we are working on GA release, so please checkout the latest possible. If trouble reproduces again, I will take a deep look at the problem.

**Comment by sujithre** [comment]
@DeagleGross  Sure :) I will use it and let you know

**Comment by sujithre** [comment]
its still the same 

The version I am using is 1.0.0-beta.5.
 
The command I have used for the Extractor is like below
 
dotnet run extract --sourceApimName XXX --resourceGroup Demo --destinationApimName XXX  --fileFolder C:\ApimAzure\Output
 
this will extract all the api Templates for in my Source API.
 
I pushed this into my Repo and configured the devops pipeline as said like below

![image](https://user-images.githubusercontent.com/25592014/172290727-5506753d-d16f-4e5b-a463-c17cee8029f2.png)


https://azure.github.io/apim-lab/apim-lab/8-devops/apimanagement-8.html

**Comment by dannythomas13** [comment]
@sujithre If you review the RG deployment within Azure Portal do you see the following?

Note that Microsoft.ApiManagement/service/apis and Microsoft.ApiManagement/service/products/apis prior to this succeeds

{
"status": "Failed",
"error": {
"code": "InternalServerError",
"message": "Request processing failed due to internal error."
}
}

Type:
Microsoft.ApiManagement/service/apis/schemas

**Comment by sujithre** [comment]
Hi@dannythomas13 ,

Yes I did see that Error for the Schemas. it will just keep trying there and eventually it gets timed out.

**Comment by BouvetMarius** [comment]
Hi @sujithre 
I think you have the same issue as i had.

Check out #761, and check if my PR #762 will fix your issue

**Comment by f-alizada** [comment]
Hello @sujithre.
Thank you for reporting the issue. 
From the issue comments it seems that the problem was the same as in #756, deployment of Api Schemas was failing with internal error and deployment timed out. 
The fix was merged into the main branch #770, hence issue will be closed. 
Please feel free to reopen this issue or create new one if you will face same behaviour with deploying API schemas from main branch or upcoming releases
Best regards
